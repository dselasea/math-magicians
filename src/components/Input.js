import React from 'react';

const Input = (props) => {
  const calc = { props };
  return (
    <div>
      <input type="text" id="input" value={`${calc.props.calculation.total || ''} ${calc.props.calculation.operation || ''} ${calc.props.calculation.next || ''}`} readOnly />
    </div>
  );
};

export default Input;
