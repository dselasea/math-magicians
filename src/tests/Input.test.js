import renderer from 'react-test-renderer';
import Input from '../components/Input';

const calculation = { total: 0, next: 0, operation: null };

test('Test Input Component', () => {
  const component = renderer.create(<Input calculation={calculation} />);
  const input = component.toJSON();
  expect(input).toMatchSnapshot();
});
