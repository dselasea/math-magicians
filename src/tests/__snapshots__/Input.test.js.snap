// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input Component 1`] = `
<div>
  <input
    id="input"
    readOnly={true}
    type="text"
    value="  "
  />
</div>
`;
